src/io/flutter/preview/render_server_template.txt

```diff
@@ -0,0 +1,192 @@
+import 'dart:convert';
+import 'dart:io' as io;
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+// TODO: import library to render
+
+void main() async {
+  try {
+    var binding = WidgetsFlutterBinding.ensureInitialized();
+
+    var rootWidget = new Container(); // TODO: create widget
+    var app = new _DesignerApp(rootWidget);
+
+    Map<int, Widget> registeredWidgets = {}; // TODO: use flutterDesignerWidgets
+    registeredWidgets.clear();
+
+    {
+      binding.attachRootWidget(app);
+      binding.scheduleFrame();
+      binding.handleBeginFrame(new Duration(milliseconds: 0));
+      binding.handleDrawFrame();
+    }
+
+    Element rootElement = WidgetsBinding.instance.renderViewElement;
+    Iterable<Element> allElements = collectAllElementsFrom(rootElement);
+    var allElementsMap = <Widget, Element>{};
+    for (var element in allElements) {
+      allElementsMap[element.widget] = element;
+    }
+
+    var rootInfo =
+        ElementInfo.buildHierarchy(allElementsMap, registeredWidgets);
+
+    {
+      var json = <String, Object>{};
+      rootInfo.appendToJson(json);
+      io.stdout.writeln(JSON.encode(json));
+    }
+
+    io.exit(0);
+  } catch (e, st) {
+    var json = {'exception': '$e', 'stackTrace': '$st'};
+    io.stdout.writeln(JSON.encode(json));
+    io.exit(1);
+  }
+}
+
+Iterable<Element> collectAllElementsFrom(Element rootElement) {
+  return new CachingIterable<Element>(
+      new _DepthFirstElementIterator(rootElement));
+}
+
+class ElementInfo {
+  final ElementInfo parent;
+  final int id;
+  final ui.Rect globalBounds;
+  final List<ElementInfo> children = <ElementInfo>[];
+
+  ElementInfo(this.parent, this.id, this.globalBounds) {
+    parent?.children?.add(this);
+  }
+
+  void appendToJson(Map<String, Object> json) {
+    json['$id'] = {
+      'globalBounds': {
+        'left': globalBounds.left.truncate(),
+        'top': globalBounds.top.truncate(),
+        'width': globalBounds.width.truncate(),
+        'height': globalBounds.height.truncate(),
+      }
+    };
+
+    for (int i = 0; i < children.length; i++) {
```

Comment:
Nit:
for (var child in children) {
  child.appendToJson(json);
}
of 
children.forEach
is a bit more idiomatic dart.

Reply:
ACK
I needed index in one of the previous incarnations of this code.
But not anymore.